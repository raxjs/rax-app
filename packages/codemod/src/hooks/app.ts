import * as path from 'path';
import * as fs from 'fs-extra';
import additionalContent from '../utils/additionalContent';

const libDirectory = path.join(__dirname, '../../lib');
export function beforeTransform(args, options) {
  const { files } = options;
  const transformerPath = path.join(libDirectory, 'transforms/app/project.js');
  args = args.concat(['--transform', transformerPath]);
  args.push('--app=true');
  args.push('--ignore-pattern=**/src/!(app.json|app.@(t|j)s?(x))/**');
  const rootDir = path.join(process.cwd(), files[0]);
  try {
    const pkgJSON = JSON.parse(fs.readFileSync(path.join(rootDir, 'package.json')));
    const customConfigPaths = [];
    Object.keys(pkgJSON.scripts).forEach((script) => {
      const matched = pkgJSON.scripts[script].match(/build-scripts (?:start|build) --config\s(\S*)/);
      if (Array.isArray(matched) && matched[1]) {
        const configPath = path.join(rootDir, matched[1]);
        if (!customConfigPaths.includes(configPath)) {
          customConfigPaths.push(configPath);
        }
      }
    });
    customConfigPaths.forEach((customConfigPath) => {
      args.push(`--customBuildJSON=${customConfigPath}`);
    });
    // eslint-disable-next-line no-empty
  } catch (err) {}
  return {
    args,
  };
}

export function afterTransform(options) {
  const { files } = options;
  const rootDir = path.join(process.cwd(), files[0]);

  // Rewrite tsconfig.json
  fs.writeFileSync(path.join(rootDir, 'tsconfig.json'), JSON.stringify(require('../templates/tsconfig.json'), null, 2));
  // Rewrite .eslintignore
  additionalContent(path.join(rootDir, '.eslintignore'), '.rax', path.join(libDirectory, 'templates/eslintignore.txt'));

  // Rewrite .gitignore
  additionalContent(path.join(rootDir, '.gitignore'), '.rax', path.join(libDirectory, 'templates/gitignore.txt'));

  // Rewrite prettier
  fs.writeFileSync(
    path.join(rootDir, '.prettierignore'),
    fs.readFileSync(path.join(libDirectory, 'templates/prettierignore.txt')),
  );
  fs.writeFileSync(
    path.join(rootDir, 'prettierrc.js'),
    fs.readFileSync(path.join(libDirectory, 'templates/prettierrc.js')),
  );

  // Rewrite style lint
  fs.writeFileSync(
    path.join(rootDir, '.stylelintignore'),
    fs.readFileSync(path.join(libDirectory, 'templates/stylelintignore.txt')),
  );
  fs.writeFileSync(
    path.join(rootDir, 'stylelintrc.js'),
    fs.readFileSync(path.join(libDirectory, 'templates/stylelintrc.js')),
  );
}
