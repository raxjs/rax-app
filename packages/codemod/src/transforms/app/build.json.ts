import * as fs from 'fs-extra';
import * as path from 'path';

const builtInPlugins = ['build-plugin-rax-multi-pages', 'build-plugin-rax-ssr'];

export default function (fileInfo) {
  const appJSON = JSON.parse(fs.readFileSync(path.join(process.cwd(), path.dirname(fileInfo.path), 'src/app.json')));
  let buildConfig = JSON.parse(fileInfo.source);

  const pluginCore = buildConfig?.plugins.find((plugin) => {
    return Array.isArray(plugin) && plugin[0] === 'build-plugin-rax-app';
  });
  if (!pluginCore) {
    return fileInfo.source;
  }

  let staticExport = false;
  let snapshot = false;
  let mpa = false;
  let ssr;
  let doctype;
  if (buildConfig.plugins.includes('build-plugin-rax-multi-pages')) {
    mpa = true;
  }

  const pluginCoreOptions = pluginCore[1] || {};
  buildConfig.plugins = deletePlugin(buildConfig.plugins, buildConfig.plugins.indexOf(pluginCore));
  builtInPlugins.forEach((pluginName) => {
    const index = buildConfig.plugins.indexOf(pluginName);
    if (index > -1) {
      if (pluginName === 'build-plugin-rax-ssr') {
        ssr = true;
      }
      buildConfig.plugins = deletePlugin(buildConfig.plugins, index);
    }
  });
  if (pluginCoreOptions.type === 'mpa') {
    mpa = true;
  }
  delete pluginCoreOptions.type;

  if (Object.prototype.hasOwnProperty.call(pluginCoreOptions, 'doctype')) {
    doctype = pluginCoreOptions.doctype;
    delete pluginCoreOptions.doctype;
  }

  const pwaPlugin = buildConfig.plugins.find((plugin) => {
    if (Array.isArray(plugin)) {
      return plugin[0] === 'build-plugin-rax-pwa' || plugin[0] === 'build-plugin-rax-pha';
    } else {
      return plugin === 'build-plugin-rax-pwa' || plugin === 'build-plugin-rax-pha';
    }
  });
  if (Array.isArray(pwaPlugin)) {
    const pwaPluginOptions = pwaPlugin[1];
    // snapshot
    if (Object.prototype.hasOwnProperty.call(pwaPluginOptions, 'snapshot')) {
      snapshot = pluginCoreOptions.snapshot;
    }
    // serviceWorker
    if (pwaPluginOptions.serviceWorker) {
      console.error('rax-app 3.x 暂不支持 serviceWorker');
    }
  }
  if (pwaPlugin) {
    buildConfig.plugins = deletePlugin(buildConfig.plugins, buildConfig.plugins.indexOf(pwaPlugin));
  }

  // static export
  if (Object.prototype.hasOwnProperty.call(pluginCoreOptions, 'staticExport')) {
    staticExport = pluginCoreOptions.staticExport;
    delete pluginCoreOptions.staticExport;
  }

  if (!Object.prototype.hasOwnProperty.call(buildConfig, 'inlineStyle')) {
    buildConfig.inlineStyle = true;
  }
  buildConfig = {
    ...pluginCoreOptions,
    ...buildConfig,
  };

  buildConfig.targets.forEach((target) => {
    switch (target) {
      // web config
      case 'web':
        buildConfig.web = {};
        if (staticExport) {
          buildConfig.web.staticExport = staticExport;
        }
        if (snapshot) {
          buildConfig.web.snapshot = snapshot;
        }
        if (mpa) {
          buildConfig.web.mpa = true;
        }
        if (doctype !== undefined) {
          buildConfig.web.doctype = doctype;
        }
        if (ssr !== undefined) {
          buildConfig.web.ssr = ssr;
        }
        if (Object.prototype.hasOwnProperty.call(appJSON, 'hydrate')) {
          buildConfig.web.hydrate = appJSON.hydrate;
        }
        break;
      // weex config
      case 'weex':
        if (mpa) {
          buildConfig.weex = {
            mpa: true,
          };
        }
        break;
      // miniapp
      case 'miniapp':
      case 'wechat-miniprogram':
        if (!buildConfig[target]) {
          buildConfig[target] = {
            buildType: 'compile',
          };
        }
        break;
      default:
    }
  });

  return JSON.stringify(buildConfig, null, 2);
}

function deletePlugin(plugins, index) {
  return [...plugins.slice(0, index), ...plugins.slice(index + 1, plugins.length)];
}
