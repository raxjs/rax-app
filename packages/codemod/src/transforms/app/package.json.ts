const depDeleteList = ['rax-app'];
const devDepDeleteList = [
  '@alib/build-scripts',
  'babel-eslint',
  'build-plugin-rax-app',
  'eslint-config-rax',
  'eslint-plugin-import',
  'eslint-plugin-module',
  'eslint-plugin-react',
  'build-plugin-rax-multi-pages',
  'build-plugin-rax-ssr',
  'build-plugin-rax-pwa',
];

const devAddDepList = [
  {
    name: '@types/rax',
    version: '^1.0.0',
  },
  {
    name: '@iceworks/spec',
    version: '^1.0.0',
  },
  {
    name: 'prettier',
    version: '^2.0.0',
  },
  {
    name: 'stylelint',
    version: '^13.0.0',
  },
  {
    name: 'rax-app',
    version: '^3.0.0',
  },
];

const updateDevList = [
  {
    name: '@ali/build-plugin-rax-faas',
    version: '^4.0.0',
  },
  {
    name: '@ali/build-plugin-rax-app-def',
    version: '^3.0.0',
  },
  {
    name: 'build-plugin-rax-compat-react',
    version: '^1.0.0',
  },
];

const extraScripts = [
  {
    name: 'eslint',
    command: 'eslint --ext .js,.jsx ./',
  },
  {
    name: 'stylelint',
    command: 'stylelint "**/*.{css,scss,less}"',
  },
  {
    name: 'prettier',
    command: 'prettier **/* --write',
  },
  {
    name: 'lint',
    command: 'npm run eslint && npm run stylelint',
  },
];

export default function (fileInfo) {
  const pkg = JSON.parse(fileInfo.source);
  depDeleteList.forEach((pkgName) => delete pkg.dependencies[pkgName]);
  devDepDeleteList.forEach((pkgName) => delete pkg.devDependencies[pkgName]);
  devAddDepList.forEach(({ name, version }) => {
    pkg.devDependencies[name] = version;
  });
  updateDevList.forEach(({ name, version }) => {
    if (pkg.devDependencies[name]) {
      pkg.devDependencies[name] = version;
    }
  });

  Object.keys(pkg.scripts).forEach((command) => {
    pkg.scripts[command] = pkg.scripts[command]
      .replace('build-scripts build', 'rax-app build')
      .replace('build-scripts start', 'rax-app start');
  });

  extraScripts.forEach(({ name, command }) => {
    if (!pkg.scripts[name]) {
      pkg.scripts[name] = command;
    }
  });

  return JSON.stringify(pkg, null, 2);
}
