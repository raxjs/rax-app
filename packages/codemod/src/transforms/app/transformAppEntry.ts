export default function (fileInfo, api) {
  const { jscodeshift } = api;
  const ast = jscodeshift(fileInfo.source);
  ast
    .find(jscodeshift.ImportDeclaration)
    .filter((path) => path.value.source.value === './app.json')
    .forEach((path) => {
      path.prune();
    });

  ast
    .find(jscodeshift.Identifier)
    .filter((path) => path.value.name === 'appConfig')
    .remove();

  return ast.toSource();
}
