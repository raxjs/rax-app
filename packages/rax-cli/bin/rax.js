#!/usr/bin/env node

const updateNotifier = require('update-notifier');
const fs = require('fs');
const fse = require('fs-extra');
const path = require('path');
const inquirer = require('inquirer');
const chalk = require('chalk');
const semver = require('semver');
const kebabCase = require('lodash.kebabcase');
const { checkAliInternal } = require('ice-npm-utils');
const argv = require('minimist')(process.argv.slice(2));
const { downloadAndGenerateProject } = require('@iceworks/generate-project');
const { generateMaterial, downloadMaterialTemplate } = require('@iceworks/generate-material');

const config = require('./config');
const pkg = require('../package.json');

let projectName = '';

// notify package update
updateNotifier({ pkg }).notify();

// Check node version
if (!semver.satisfies(process.version, '>=8')) {
  const message = `You are currently running Node.js ${
    chalk.red(process.version)}.\n` +
    '\n' +
    'Rax runs on Node 6.0 or newer. There are several ways to ' +
    'upgrade Node.js depending on your preference.\n' +
    '\n' +
    'nvm:       nvm install node && nvm alias default node\n' +
    'Homebrew:  brew install node\n' +
    'Installer: download the Mac .pkg from https://nodejs.org/\n';

  console.log(message);
  process.exit(1);
}


/**
 * check version
 * usage: rax -v or rax --version
 */
if (argv._.length === 0 && (argv.v || argv.version)) {
  console.log(`rax-cli: ${pkg.version}`);
  try {
    const RAX_PACKAGE_JSON_PATH = path.resolve(
      process.cwd(),
      'node_modules',
      'rax',
      'package.json',
    );
    console.log(`rax: ${require(RAX_PACKAGE_JSON_PATH).version}`);
  } catch (e) {
    console.log('rax: n/a - not inside a Rax project directory');
  }
  process.exit();
}

function printHelp() {
  console.log(`Usage: rax <command> [options]
Options:
  -v, --version                              output the version number
  -h, --help                                 output usage information

Commands:
  init <app-name>                            generate project directory based on templates

Run rax <command> --help for detailed usage of given command.
`);
}

function printInitHelp() {
  console.log(`Usage: rax init [options] <app-name>

Options:
  -v, --verbose                               show project init details
`);
}

if (argv._.length === 0 && (argv.h || argv.help)) {
  printHelp();
  process.exit();
}

// check commands
const commands = argv._;
let createInCurrent = true;

if (commands.length === 0) {
  printHelp();
  process.exit(1);
}
switch (commands[0]) {
  case 'init':
    if (argv.h || argv.help) {
      printInitHelp();
      process.exit();
    }

    init(commands[1], argv.v || argv.verbose, argv.t || argv.template, argv.s || argv['skip-ask']);
    break;
  default:
    console.error(
      chalk.red(`Command \`${commands[0]}\` unrecognized.`),
    );
    printHelp();
    process.exit(1);
    break;
}

/**
 * rax init
 */
async function init(name, verbose, template, skipAsk) {
  projectName = name;
  createInCurrent = !projectName;
  if (!projectName) {
    projectName = process.cwd().split(path.sep).pop();
  }

  let answers = {};

  if (!skipAsk) {
    answers = await askProjectInformaction(template);
  }

  await createProject(kebabCase(projectName), verbose, template, answers);
}

function askProjectInformaction(appTemplate) {
  const rootDir = path.resolve(projectName);
  const conflictFiles = ['src', 'build.json', 'package.json'];

  /**
   * Check whether contains conflict files
   * @param  {String} targetDir
   * @return {Boolean}
   */
  const containConflictFile = (targetDir) => {
    return conflictFiles.some(filename => fs.existsSync(path.join(targetDir, filename)));
  };

  let prompts = config.getPromptQuestion(appTemplate);
  if (containConflictFile(rootDir)) {
    prompts = [
      {
        type: 'confirm',
        name: 'shouldInputNewProjectName',
        message: `The directory ${projectName} contains files that could conflict:\n\n${conflictFiles.join('\n')}\n\nEither try using a new directory name, or still use the directory ${projectName}.`,
        default: true
      },
      {
        type: 'input',
        name: 'projectName',
        message: 'Please input a new directory name.',
        when(answers) {
          return answers.shouldInputNewProjectName;
        },
        validate: (newName) => {
          const newRootDir = path.resolve(newName);
          if (newName === projectName) {
            return 'Same as the original name, please change another name.';
          } else if (containConflictFile(newRootDir)) {
            return 'This directory also contains files that could conflict, please change another name.';
          } else if (!newName) {
            return 'Please enter a name that cannot be empty.';
          }
          projectName = newName;
          return true;
        }
      }
    ].concat(prompts);
  }

  return inquirer.prompt(prompts);
}

async function createProject(name, verbose, template, userAnswers) {
  const projectName = name;

  let rootDir = process.cwd();
  if (!createInCurrent) {
    rootDir = path.resolve(name);
    if (!fs.existsSync(rootDir)) {
      fs.mkdirSync(rootDir);
    }
    process.chdir(rootDir);
  }

  const { projectType = 'app', projectTargets, appType, languageType } = userAnswers;
  const registry = 'https://registry.npm.taobao.org';

  console.log(
    'Creating a new Rax project in',
    rootDir
  );

  if (projectType === 'app') {
    const appTemplate = template || (languageType === 'ts' ? '@rax-materials/scaffolds-app-ts' : '@rax-materials/scaffolds-app-js');
    await downloadAndGenerateProject(
      rootDir,
      appTemplate,
      null,
      registry,
      null,
      {
        targets: projectTargets,
        mpa: appType === 'mpa'
      }
    );
  } else {
    const typeToTemplate = {
      component: {
        ts: '@icedesign/template-rax',
        js: '@icedesign/template-rax',
      },
      api: '@icedesign/template-rax-api',
      plugin: '@icedesign/template-rax-miniapp-plugin'
    };

    const tempDir = path.join(rootDir, '.tmp');
    const materialTemplate = typeToTemplate[projectType][languageType] ? typeToTemplate[projectType][languageType] : typeToTemplate[projectType];

    await downloadMaterialTemplate(tempDir, materialTemplate, registry);
    await generateMaterial({
      rootDir,
      materialTemplateDir: tempDir,
      templateOptions: {
        npmName: 'rax-example',
        projectTargets,
      },
      materialType: 'component'
    });
    await fse.remove(tempDir);
  }

  console.log(chalk.white.bold('To run your app:'));
  if (!createInCurrent) {
    console.log(chalk.white(`   cd ${projectName}`));
  }

  const isAliInternal = await checkAliInternal();
  let npmCommand = 'npm';
  let explanation = '';
  if (isAliInternal) {
    npmCommand = 'tnpm';
    explanation = 'Detected that you are an Alibaba user, DEF plugin has been added!\n\n';
  }

  console.log(chalk.white(`   ${npmCommand} install`));
  console.log(chalk.white(`   ${npmCommand} start`));
  console.log(chalk.white(`${explanation}We have prepared develop toolkit for you. \nSee: https://marketplace.visualstudio.com/items?itemName=iceworks-team.iceworks`));
}
