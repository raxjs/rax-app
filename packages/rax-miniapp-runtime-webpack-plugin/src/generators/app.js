const { resolve, relative, extname } = require('path');
const { readFileSync } = require('fs-extra');
const ejs = require('ejs');
const adapter = require('../adapter');
const { MINIAPP } = require('../constants');
const addFileToCompilation = require('../utils/addFileToCompilation');
const getAssetPath = require('../utils/getAssetPath');
const adjustCSS = require('../utils/adjustCSS');

function generateAppJS(
  compilation,
  commonAppJSFilePaths,
  { target, command, rootDir }
) {
  const appJsTmpl = readFileSync(
    resolve(rootDir, 'templates', 'app.js.ejs'),
    'utf-8'
  );
  const appJsContent = ejs.render(appJsTmpl, {
    init: `function init(window, document) {${commonAppJSFilePaths
      .map(
        filePath =>
          `require('${getAssetPath(
            relative(target, filePath),
            'app.js'
          )}')(window, document)`
      )
      .join(';')}}`,
    isMiniApp: target === MINIAPP
  });
  addFileToCompilation(compilation, {
    filename: 'app.js',
    content: appJsContent,
    target,
    command,
  });
}

function generateAppCSS(compilation, { target, command, rootDir }) {
  // Add default css file to compilation
  const defaultCSSTmpl = adjustCSS(readFileSync(
    resolve(rootDir, 'templates', 'default.css.ejs'),
    'utf-8'
  ));
  addFileToCompilation(compilation, {
    filename: `default.${adapter[target].css}`,
    content: defaultCSSTmpl,
    target,
    command,
  });

  delete compilation.assets[`${target}/app.css`];

  let content = '@import "./default";';

  Object.keys(compilation.assets).forEach(asset => {
    if (asset.indexOf(`${target}/index`) > -1 && extname(asset) === '.css') {
      content += `@import "./${relative(target, asset)}";`;
      delete compilation.assets[asset];
    }
  });

  addFileToCompilation(compilation, {
    filename: `app.${adapter[target].css}`,
    content,
    target,
    command,
  });
}

module.exports = {
  generateAppJS,
  generateAppCSS
};
