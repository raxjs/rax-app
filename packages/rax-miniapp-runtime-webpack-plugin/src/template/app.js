/* INIT_FUNCTION */
/* global appConfig,App,getCurrentPages */
const LIFE_CYCLE_METHODS = [
  'onLaunch',
  'onShow',
  'onHide',
  'onError',
  'onPageNotFound',
];
const extraConfig = {};
for (const key in appConfig) {
  if (LIFE_CYCLE_METHODS.indexOf(key) === -1) extraConfig[key] = appConfig[key];
}

// eslint-disable-next-line new-cap
App({
  onLaunch(options) {
    if (appConfig.onLaunch) appConfig.onLaunch.call(this, options);
  },
  onShow(options) {
    if (appConfig.onShow) appConfig.onShow.call(this, options);
  },
  onHide() {
    if (appConfig.onHide) appConfig.onHide.call(this);
  },
  onError(err) {
    // Support error event of window
    const pages = getCurrentPages() || [];
    const currentPage = pages[pages.length - 1];
    if (currentPage && currentPage.window) {
      currentPage.window.$$trigger('error', {
        event: err,
      });
    }

    if (appConfig.onError) appConfig.onError.call(this, err);
  },
  onPageNotFound(options) {
    if (appConfig.onPageNotFound) appConfig.onPageNotFound.call(this, options);
  },

  ...extraConfig,
});
