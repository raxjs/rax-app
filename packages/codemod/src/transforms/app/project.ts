import * as path from 'path';
import transformBuildConfig from './build.json';
import transformPkgJSON from './package.json';
import transformAppJSON from './app.json';
import transformEslint from './eslint';
import transformAppEntry from './transformAppEntry';

export default function (fileInfo, api, options) {
  let source;
  switch (fileInfo.path) {
    case 'package.json':
      source = transformPkgJSON(fileInfo);
      break;
    case 'build.json':
      source = transformBuildConfig(fileInfo);
      break;
    case 'src/app.json':
      source = transformAppJSON(fileInfo);
      break;
    case 'src/app.js':
    case 'src/app.jsx':
    case 'src/app.ts':
    case 'src/app.tsx':
      source = transformAppEntry(fileInfo, api);
      break;
    case '.eslintrc.js':
      source = transformEslint();
      break;
    default:
      if (options.customBuildJSON) {
        const absolutePath = path.resolve(process.cwd(), fileInfo.path);
        if (Array.isArray(options.customBuildJSON) && options.customBuildJSON.includes(absolutePath)) {
          source = transformBuildConfig(fileInfo);
        } else if (options.customBuildJSON === absolutePath) {
          source = transformBuildConfig(fileInfo);
        }
      } else {
        source = fileInfo.source;
      }
  }

  return source;
}
