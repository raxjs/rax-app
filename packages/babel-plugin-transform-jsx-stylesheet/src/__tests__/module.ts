import * as syntaxJSX from 'babel-plugin-syntax-jsx';
import { transform } from 'babel-core';
import jSXStylePlugin from '../index';

export default function getTransformCode(code, opts?) {
  return transform(code, {
    plugins: [
      [jSXStylePlugin, opts],
      syntaxJSX,
    ],
  }).code;
}

describe('css module', () => {
  it('should transform css module', () => {
    expect(getTransformCode(`
  import { createElement, render } from 'rax';
  import styles from './app.css';

  render(<div className={styles.header} />);
  `)).toBe(`
import { createElement, render } from 'rax';
import styles from './app.css';

render(<div style={styles.header} />);`);
  });

  it('should transform css when `forceEnable` is true and is not `*.module.css`', () => {
    expect(getTransformCode(`
  import { createElement, render } from 'rax';
  import styles from './app.css';

  render(<div className={styles.header} />);
  `, { forceEnableCSS: true })).toBe(`
import { createElement, render } from 'rax';
import styles from './app.css';

render(<div style={styles.header} />);`);
  })

  it('should not transform css when `forceEnable` is true and is `*.module.css`', () => {
    expect(getTransformCode(`
  import { createElement, render } from 'rax';
  import styles from './app.module.css';

  render(<div className={styles.header} />);
  `, { forceEnableCSS: true })).toBe(`
import { createElement, render } from 'rax';
import styles from './app.module.css';

render(<div className={styles.header} />);`);
  })

  it('should not transform css when `forceEnableCSS` is true and module or not module simultaneously exist', () => {
    expect(getTransformCode(`
import { createElement, render } from 'rax';
import app from './app.css';
import styles from './app.module.css';

render(<div className={styles.header} style={app.header} />);
`, { forceEnableCSS: true })).toBe(`
import { createElement, render } from 'rax';
import app from './app.css';
import styles from './app.module.css';

render(<div className={styles.header} style={app.header} />);`);
  });

  it('should transform code correctly when `retainClassName` is true', () => {
    expect(getTransformCode(`
import { createElement, render } from 'rax';
import styles from './app.css';

render(<div className={styles.header} />);
`, { retainClassName: true })).toBe(`
import { createElement, render } from 'rax';
import styles from './app.css';

render(<div style={styles.header} />);`);
  });
  expect(getTransformCode(`
import { createElement, render } from 'rax';
import styles from './app.css';

render(<div className={styles.header} />);
`, { retainClassName: true })).toMatchSnapshot();
});
