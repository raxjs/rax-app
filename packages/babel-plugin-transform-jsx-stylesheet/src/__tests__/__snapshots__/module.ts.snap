// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`css module should transform code correctly when \`retainClassName\` is true 1`] = `
"
import { createElement, render } from 'rax';
import styles from './app.css';

render(<div style={styles.header} />);"
`;

exports[`css module should transform code correctly when \`retainClassName\` is true and use different attr 1`] = `
"
import { createElement, render } from 'rax';
import styles from './app.css';

render(<div style={Object.assign({}, styles.header, styles.container)} />);"
`;
